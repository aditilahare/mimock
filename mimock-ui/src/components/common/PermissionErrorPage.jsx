import React from 'react';
import { FaUserLock } from 'react-icons/fa';
import {
	PermissionErrorPageWrapper,
	ErrorLabel,
	IconWrapper,
} from './PermissionErrorPage.style.js';

export default function PermissionErrorPage() {
	return (
		<PermissionErrorPageWrapper data-testid='permission-error-page'>
			<IconWrapper>
				<FaUserLock />
			</IconWrapper>
			<ErrorLabel data-testid='permission-error-label'>
				User not permitted to perform the action
			</ErrorLabel>
		</PermissionErrorPageWrapper>
	);
}
