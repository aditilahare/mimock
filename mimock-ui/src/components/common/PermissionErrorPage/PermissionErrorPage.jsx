import React from 'react';
import { FaUserLock } from 'react-icons/fa';
import { IconButtonWithLabel, IconButtonVariants } from 'styles/Button';
import { routes } from 'constants/routes';
import {
	PermissionErrorPageWrapper,
	ErrorLabel,
	IconWrapper,
	HomeLink,
} from './PermissionErrorPage.style.js';

export default function PermissionErrorPage() {
	return (
		<PermissionErrorPageWrapper data-testid='permission-error-page'>
			<IconWrapper>
				<FaUserLock />
			</IconWrapper>
			<ErrorLabel data-testid='permission-error-label'>
				User not permitted to perform the action
			</ErrorLabel>
			<HomeLink
				data-testid='permission-error-home-link'
				to={routes.mocks.path}
				replace={true}
			>
				<IconButtonWithLabel
					label='Go back to Dashboard'
					variant={IconButtonVariants.HomeButton}
				/>
			</HomeLink>
		</PermissionErrorPageWrapper>
	);
}
