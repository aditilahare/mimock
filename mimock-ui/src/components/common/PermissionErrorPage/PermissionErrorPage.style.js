import tw from 'tailwind-styled-components';

export const PermissionErrorPageWrapper = tw.div`
	w-full
    h-full
    flex
	flex-col
    bg-gray-200
    opacity-75
    z-999
    fixed
    top-0
    bottom-0
    left-0
    right-0
	text-8xl
	text-gray-600
	justify-center
	items-center
`;

export const IconWrapper = tw.div`
	w-full
	flex
	justify-center
`;

export const ErrorLabel = tw.span`
	text-center
	text-2xl
	text-gray-600
	w-full
`;
