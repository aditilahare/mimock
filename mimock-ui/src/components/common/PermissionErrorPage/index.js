import PermissionErrorPage from './PermissionErrorPage';

export default PermissionErrorPage;
