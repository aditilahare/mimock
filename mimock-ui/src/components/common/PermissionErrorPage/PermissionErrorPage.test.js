import React from 'react';
import PermissionErrorPage from './PermissionErrorPage.jsx';
import { render } from '@testing-library/react';

describe('PermissionErrorPage', () => {
	it('should render PermissionErrorPage', async () => {
		const { getByTestId, container } = await render(<PermissionErrorPage />);

		expect(getByTestId('permission-error-page')).toBeInTheDocument();
		expect(getByTestId('permission-error-label')).toBeInTheDocument();
		expect(getByTestId('permission-error-label').textContent).toBe(
			'User not permitted to perform the action'
		);

		expect(container).toMatchSnapshot();
	});
});
