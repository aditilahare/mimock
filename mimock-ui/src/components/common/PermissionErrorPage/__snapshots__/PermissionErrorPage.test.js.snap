// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PermissionErrorPage should render PermissionErrorPage 1`] = `
<div>
  <div
    class="w-full h-full flex flex-col bg-gray-200 opacity-75 z-999 fixed top-0 bottom-0 left-0 right-0 text-8xl text-gray-600 justify-center items-center"
    data-testid="permission-error-page"
  >
    <div
      class="w-full flex justify-center"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224 256A128 128 0 1 0 96 128a128 128 0 0 0 128 128zm96 64a63.08 63.08 0 0 1 8.1-30.5c-4.8-.5-9.5-1.5-14.5-1.5h-16.7a174.08 174.08 0 0 1-145.8 0h-16.7A134.43 134.43 0 0 0 0 422.4V464a48 48 0 0 0 48 48h280.9a63.54 63.54 0 0 1-8.9-32zm288-32h-32v-80a80 80 0 0 0-160 0v80h-32a32 32 0 0 0-32 32v160a32 32 0 0 0 32 32h224a32 32 0 0 0 32-32V320a32 32 0 0 0-32-32zM496 432a32 32 0 1 1 32-32 32 32 0 0 1-32 32zm32-144h-64v-80a32 32 0 0 1 64 0z"
        />
      </svg>
    </div>
    <span
      class="text-center text-2xl text-gray-600 w-full"
      data-testid="permission-error-label"
    >
      User not permitted to perform the action
    </span>
    <a
      class="my-4"
      data-testid="permission-error-home-link"
      href="/mimock-ui/mocks"
    >
      <button
        class="flex items-center justify-center gap-1 px-9 py-2 shadow-md rounded-sm text-center text-base font-semibold outline-none cursor-pointer hover:brightness-110 transition duration-100 text-white w-fit bg-indigo-600 dark:bg-indigo-800"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 576 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
          />
        </svg>
        Go back to Dashboard
      </button>
    </a>
  </div>
</div>
`;
